library/std/src/process.rs

```diff
@@ -1725,6 +1725,48 @@ impl ExitCode {
     /// return the same codes (but will also `eprintln!` the error).
     #[stable(feature = "process_exitcode", since = "1.60.0")]
     pub const FAILURE: ExitCode = ExitCode(imp::ExitCode::FAILURE);
+
+    /// Exit the current process with the given `ExitCode`.
+    ///
+    /// Note that this has the same caveats as [`process::exit()`][exit], namely that this function
+    /// terminates the process immediately, so no destructors on the current stack or any other
+    /// thread's stack will be run. If a clean shutdown is needed, it is recommended to simply
+    /// return this ExitCode from the `main` function, as demonstrated in the [type
+    /// documentation](#examples).
+    ///
+    /// # Differences from `process::exit()`
+    ///
+    /// `process::exit()` accepts any `i32` value as the exit code for the process; however, there
+    /// are platforms that only use a subset of that value (see [`process::exit` platform-specific
+    /// behavior][exit#platform-specific-behavior]). `ExitCode` exists because of this; only
+    /// `ExitCode`s that are valid on all platforms can be created, so those problems don't exist
```

Comment:
I think we need to hedge here a bit. We already have some inconsistencies between platforms for just the `u8` range of `ExitCode`, I think some unixes only use 7 of the bits, and quite a few platforms only support `SUCCESS`/`FAILURE`.

```suggestion
    /// `ExitCode`s that are supported by a majority of our platforms can be created, so those problems don't exist
```

Reply:
I also made the wording on the next line "so these problems don't exist (as much) [...]".